skippkg-finder.py

```diff
@@ -0,0 +1,331 @@
+            return True
+        if pkg.startswith('sle15') or\
+                pkg.startswith('sles15') or\
+                pkg.startswith('suse-migration') or\
+                pkg.startswith('migrate') or\
+                pkg.startswith('kernel-livepatch') or\
+                pkg.startswith('patterns') or\
+                pkg.startswith('supportutils-plugin') or\
+                pkg.startswith('lifecycle-data-sle') or\
+                pkg.startswith('sca-patterns') or\
+                pkg.startswith('susemanager-') or\
+                pkg.startswith('kiwi-template') or\
+                pkg.startswith('desktop-data'):
+            return True
+        if pkg.endswith('bootstrap') or pkg.endswith('-caasp') or\
+                pkg.endswith('-sle'):
+            return True
+        if pkg == 'suse-build-key' or pkg == 'suse-hpc' or\
+                pkg == 'llvm' or\
+                pkg == 'txt2tags' or\
+                pkg == 'zypper-search-packages-plugin' or\
+                pkg == 'python-pymemcache' or\
+                pkg == 'python-ibus':
+            return True
+        return False
+
+    def get_packagelist(self, project, by_project=True):
+        """
+        Return the list of package's info of a project.
+        If the latest package is from an incident then returns incident
+        package.
+        """
+
+        pkglist = {}
+        packageinfo = {}
+        query = {'expand': 1}
+        root = ET.parse(http_GET(makeurl(self.apiurl, ['source', project],
+                                 query=query))).getroot()
+        for i in root.findall('entry'):
+            pkgname = i.get('name')
+            orig_project = i.get('originproject')
+            is_incidentpkg = False
+            if pkgname.startswith('000') or\
+                    pkgname.startswith('_') or\
+                    pkgname.startswith('patchinfo.') or\
+                    pkgname.startswith('skelcd-') or\
+                    pkgname.startswith('installation-images') or\
+                    pkgname.endswith('-mini'):
+                continue
+            # ugly hack for go1.x incidents as the name would be go1.x.xxx
+            if '.' in pkgname and re.match(r'[0-9]+$', pkgname.split('.')[-1]) and \
+                    orig_project.startswith('SUSE:') and orig_project.endswith(':Update'):
+                is_incidentpkg = True
+                if pkgname.startswith('go1') or pkgname.startswith('bazel0') or \
+                        pkgname.startswith('dotnet') or pkgname.startswith('ruby2'):
+                    if not (pkgname.count('.') > 1):
+                        is_incidentpkg = False
+
+            # if an incident then update the package origin info
+            if is_incidentpkg:
+                orig_name = re.sub(r'\.[0-9]+$', '', pkgname)
+                incident_number = int(pkgname.split('.')[-1])
+                if orig_name in pkglist and pkglist[orig_name]['Project'] == orig_project:
```

Comment:
BTW the code here is comparing incident number of the same source package from the matched `originproject`, the newest one has picked, avoiding spend time to poke package's meta for incident package name.